src/Service/PromotionsService.php

```diff
@@ -0,0 +1,72 @@
+<?php
+
+namespace BBC\ProgrammesPagesService\Service;
+
+use BBC\ProgrammesPagesService\Cache\CacheInterface;
+use BBC\ProgrammesPagesService\Data\ProgrammesDb\EntityRepository\PromotionRepository;
+use BBC\ProgrammesPagesService\Domain\ApplicationTime;
+use BBC\ProgrammesPagesService\Domain\Entity\CoreEntity;
+use BBC\ProgrammesPagesService\Mapper\ProgrammesDbToDomain\PromotionMapper;
+
+class PromotionsService extends AbstractService
+{
+    /** @var PromotionRepository */
+    protected $repository;
+
+    public function __construct(
+        PromotionRepository $repository,
+        PromotionMapper $mapper,
+        CacheInterface $cache
+    ) {
+        parent::__construct($repository, $mapper, $cache);
+    }
+
+    /**
+     * @return Promotion[]
+     */
+    public function findActivePromotionsByEntity(
+        CoreEntity $context,
+        ?int $limit = self::DEFAULT_LIMIT,
+        int $page = self::DEFAULT_PAGE,
+        $ttl = CacheInterface::NORMAL
+    ): array {
+        $key = $this->cache->keyHelper(__CLASS__, __FUNCTION__, $context->getDbId(), $limit, $page, $ttl);
+
+        return $this->cache->getOrSet(
+            $key,
+            $ttl,
+            function () use ($context, $limit, $page) {
+                $dbEntities = $this->repository->findActivePromotionsByContext(
+                    $context->getDbId(),
+                    ApplicationTime::getTime(),
+                    $limit,
+                    $this->getOffset($limit, $page)
+                );
+
+                $promotions = $this->mapManyEntities($dbEntities);
+                return $this->groupPromotionsByType($promotions);
+            }
+        );
+    }
+
+    /**
+     * @param Promotion[] $promotions
```

Comment:
arg....I really hate this limitation on PHP, is still so little descriptive.....
I remember in some of my previous companies we were using swagger to be more descriptives. 
https://apihandyman.io/writing-openapi-swagger-specification-tutorial-part-4-advanced-data-modeling/#arrays-size-and-uniqueness

I know is more verbose, and is not a PHP standard, but swagger is also wide used and is supported by many tools. I'm dropping this thought because some notation is still quite confuse